### PR TITLE
Azerbaijan large update

### PR DESCRIPTION
**Pull request by KJhellico**
- update holidays
- observed dates calculation completely rewritten

### CODE REVIEW

**Comment by KJhellico** [comment]
Seems something wrong with test workflow on Github

**Comment by arkid15r** [comment]
> Seems something wrong with test workflow on Github

Yes, the `coverage` is broken. This https://github.com/dr-prodigy/python-holidays/pull/862 might help.

File: test/countries/test_azerbaijan.py [hunk 1]
```diff
@@ -9,55 +9,200 @@
...
+    def test_country_aliases(self):
+        self.assertCountryAliases(Azerbaijan, AZ, AZE)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Azerbaijan(years=1989))
+
+    def test_new_year(self):
+        self.assertHoliday(*[f"{year}-01-01" for year in range(1990, 2050)])
+        self.assertHoliday(*[f"{year}-01-02" for year in range(2006, 2050)])
+        self.assertNoHoliday(*[f"{year}-01-02" for year in range(1990, 2006)])
```
Comment (arkid15r) [inline test/countries/test_azerbaijan.py:29]:
```suggestion
        self.assertHoliday(f"{year}-01-01" for year in range(1990, 2050))
        self.assertHoliday(f"{year}-01-02" for year in range(2006, 2050))
        self.assertNoHoliday(f"{year}-01-02" for year in range(1990, 2006))
```

File: test/countries/test_azerbaijan.py [hunk 2]
```diff
@@ -9,55 +9,200 @@
...
+            "2022-01-03",
+            "2022-01-04",
+            "2022-03-25",
+            "2022-05-30",
+            "2022-06-27",
+            "2022-07-11",
+            "2022-07-12",
+        )
+        self.assertHoliday(Azerbaijan(observed=True), *observed_holidays)
+        self.assertNoHoliday(Azerbaijan(observed=False), *observed_holidays)
```
Comment (arkid15r) [inline test/countries/test_azerbaijan.py:145]:
```suggestion
        self.assertHoliday(Azerbaijan(observed=True), observed_holidays)
        self.assertNoHoliday(Azerbaijan(observed=False), observed_holidays)
```

**Review by arkid15r — COMMENTED** [review]
The recent `TestCase` updates allow doing this now:

File: test/countries/test_azerbaijan.py (hunk 1 above)
Comment (KJhellico) [reply test/countries/test_azerbaijan.py]: Done.

**Pull request merged**